### PR TITLE
Handle pending and error states for rubric ai evaluation

### PR DESCRIPTION
**Pull request by Nokondi**
Updated RubricSettings and rubrics_controller components to handle error messages for PII, profanity, and general failures.

General Error Message
image

PII Error Message
image

Profanity Filter Error Message
image

### CODE REVIEW

File: apps/src/templates/rubrics/RubricSettings.jsx [hunk 1]
```diff
@@ -67,10 +73,18 @@ export default function RubricSettings({
...
+            console.log(data);
             if (!data.attempted) {
               setStatus(STATUS.NOT_ATTEMPTED);
             } else if (data.lastAttemptEvaluated) {
               setStatus(STATUS.ALREADY_EVALUATED);
+            } else if (data.status === 1000) {
+              setStatus(STATUS.ERROR);
+            } else if (data.status === 1001) {
+              setStatus(STATUS.PII_ERROR);
+            } else if (data.status === 1002) {
```
Comment (davidsbailey) [inline apps/src/templates/rubrics/RubricSettings.jsx]: please use the js version of these shared constants, here and below. for example: 
* https://github.com/code-dot-org/code-dot-org/blob/a594ea39f828105724c9d6fb486a2dbb74b6b149/apps/src/templates/rubrics/AiAssessmentBox.jsx#L12
* https://github.com/code-dot-org/code-dot-org/blob/a594ea39f828105724c9d6fb486a2dbb74b6b149/apps/src/templates/rubrics/AiAssessmentBox.jsx#L27

File: apps/src/templates/rubrics/RubricSettings.jsx [hunk 2]
```diff
@@ -91,9 +105,17 @@ export default function RubricSettings({
             setStatus(STATUS.ERROR);
           } else {
             response.json().then(data => {
-              if (data.lastAttemptEvaluated) {
+              console.log('2nd');
+              console.log(data);
```
Comment (davidsbailey) [inline apps/src/templates/rubrics/RubricSettings.jsx]: remove debug code

**Review by davidsbailey — COMMENTED** [review]
hi Mark, just a couple of comments, but I'd like to look again after you are using the shared constants.

since you are already doing the work of locating the RubricAiEvaluation and sending its status to the client, I would suggest also handling the pending state in this PR. this has 2 parts:
1. (more important) the teacher click Run AI, and we use the evaluation status (rather than the other handwavy logic about attempted and last_attempt_evaluated) to determine whether the request is pending
2. the teacher gets bored and refreshes the page while the ai eval is running. we should show them the pending state correctly on page load, rather than allowing them to click Run AI again

if you want this to be in a separate PR that is ok too -- the error state is most important because teachers won't likely get bored and refresh too often.

File: apps/src/templates/rubrics/RubricSettings.jsx (hunk 1 above)
Comment (Nokondi) [reply apps/src/templates/rubrics/RubricSettings.jsx]: fixed

File: apps/src/templates/rubrics/RubricSettings.jsx (hunk 2 above)
Comment (Nokondi) [reply apps/src/templates/rubrics/RubricSettings.jsx]: fixed

File: apps/test/unit/templates/rubrics/RubricSettingsTest.jsx [hunk 3]
```diff
@@ -170,8 +179,9 @@ describe('RubricSettings', () => {
       await act(async () => {
         await Promise.resolve();
       });
+      wrapper.update();
       expect(wrapper.find('Button').props().disabled).to.be.true;
-      expect(wrapper.text()).include(i18n.aiEvaluationStatus_pending());
+      expect(wrapper.text()).include(i18n.aiEvaluationStatus_in_progress());
```
Comment (davidsbailey) [inline apps/test/unit/templates/rubrics/RubricSettingsTest.jsx:184]: style nit: I realize you're just following an existing pattern here, but in the future, please use camelCase for js variable names where possible, e.g. `aiEvaluationStatusInProgress`. 

Most of our style rules are automatically enforced by prettier. the rest should be documented here: https://github.com/code-dot-org/code-dot-org/blob/staging/STYLEGUIDE.md

**Review by davidsbailey — APPROVED** [review]

File: apps/test/unit/templates/rubrics/RubricSettingsTest.jsx (hunk 3 above)
Comment (Nokondi) [reply apps/test/unit/templates/rubrics/RubricSettingsTest.jsx:184]: No problem. Yeah, I was just matching the pattern from the pervious strings.

**Pull request merged**